library/src/blas2/trmv_device.hpp

```diff
@@ -0,0 +1,205 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+#pragma once
+
+#include "utility.h"
+
+template <rocblas_int NB, typename T>
+__device__ void trmvn_kernel_calc(rocblas_fill     uplo,
+                                  rocblas_diagonal diag,
+                                  rocblas_int      m,
+                                  const T*         A,
+                                  rocblas_int      lda,
+                                  T*               x,
+                                  rocblas_int      incx,
+                                  T*               w)
+{
+    ptrdiff_t   tid = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
```

Comment:
Just like with all the triangular kernels I've been writing, performance can definitely be improved, but that can be saved for a later date I'm sure.

Reply:
Fully agree

Reply:
Should we create JIRA ticket for kernel optimizations? To keep track.

Reply:
Yeah that's a good idea. I see there's some specific optimization tickets already (see http://ontrack-internal.amd.com/browse/SWDEV-172731).